### PR TITLE
Kaiju2table module

### PR DESCRIPTION
**Pull request by sofstam**
## PR checklist

This PR adds the kaiju2table module #1398 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/centrifuge/kreport/main.nf [hunk 1]
```diff
@@ -12,8 +12,8 @@ process CENTRIFUGE_KREPORT {
     path db
 
     output:
-    tuple val(meta), path('*.txt')                , emit: kreport
-    path "versions.yml"                                  , emit: versions
+    tuple val(meta), path('*.txt')         , emit: kreport
```
Comment (jfy133) [inline modules/centrifuge/kreport/main.nf:15]: Better not to do this in this PR as it'll make a confusing commit message / module version description (i.e. referring to Kaiju but in the centrifuge module?)

File: modules/kaiju/kaiju2table/meta.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        e.g. [ id:'test', single_end:false ]
+  - results:
+      type: file
+      description: File containing the kaiju classification results
+      pattern: "*.{tsv}"
+  - taxon_rank:
+      type: string
+      description: |
+        Taxonomic rank to display in report
+        e.g. [ 'phylum', 'class', 'order', 'family', 'genus', 'species']
```
Comment (jfy133) [inline modules/kaiju/kaiju2table/meta.yml:29]:
```suggestion
      description: |
        Taxonomic rank to display in report
      pattern: "phylum|class|order|family|genus|species"
```

File: tests/modules/kaiju/kaiju2table/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
+          file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
+    ]
+    db    = [ [], file(params.test_data['sarscov2']['genome']['kaiju_tar_gz'], checkIfExists: true) ]
+    taxon_rank = "species"
+
+    ch_db = UNTAR ( db )
+    KAIJU_KAIJU ( input, ch_db.untar.map{ it[1] } )
+    KAIJU_KAIJU2TABLE ( KAIJU_KAIJU.out.results, ch_db.untar.map{ it[1] }, taxon_rank )
+}
```
Comment (jfy133) [inline tests/modules/kaiju/kaiju2table/main.nf:36]:
```suggestion
```

i don't think this is necessary in this case, the input to KAIJU2TABLE is the same in both tests and you don't specify any other extra parameters

File: tests/modules/kaiju/kaiju2table/test.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- name: kaiju kaiju2table test_kaiju_kaiju_paired_end
+  command: nextflow run tests/modules/kaiju/kaiju2table -entry test_kaiju_kaiju_paired_end -c tests/config/nextflow.config
+  tags:
+    - kaiju
+    - kaiju/kaiju2table
+  files:
+    - path: output/kaiju/test.txt
+      md5sum: dd8d6425d9cbfa26ce91e82ed4d4f08c
+    - path: output/kaiju/versions.yml
+      md5sum: 3704e7d4527b7f0787844142c7dafe69
```
Comment (jfy133) [inline tests/modules/kaiju/kaiju2table/test.yml:21]:
```suggestion

```

File: tests/modules/kaiju/kaiju2table/test.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+- name: kaiju kaiju2table test_kaiju_kaiju_single_end
+  command: nextflow run tests/modules/kaiju/kaiju2table -entry test_kaiju_kaiju_single_end -c tests/config/nextflow.config
+  tags:
+    - kaiju
+    - kaiju/kaiju2table
+  files:
+    - path: output/kaiju/test.txt
+      md5sum: 0d9f8fd36fcf2888296ae12632c5f0a8
+    - path: output/kaiju/versions.yml
+      md5sum: 837d886e5de4187cb9d1c8dd9f0d9c96
```
Comment (jfy133) [inline tests/modules/kaiju/kaiju2table/test.yml:10]:
```suggestion
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
overall LGTM, just suggesting some simplifications

File: modules/centrifuge/kreport/main.nf (hunk 1 above)
Comment (sofstam) [reply modules/centrifuge/kreport/main.nf:15]: I agree!

**Review by jfy133 — APPROVED** [review]
You are GO for squash and merge ;) Thanks @sofstam !

**Comment by sofstam** [comment]
I have to pull upstream first :D

**Pull request merged**